Makefile

```diff
@@ -3,37 +3,74 @@
 # Copyright 2014 The CoreOS Authors
 # Distributed under the terms of the GNU General Public License v2
 
+# used in "all" and "layout" targets
+LIBDIRS ?= lib
+
+INSTALL       ?= install
+INSTALL_DIR    = $(INSTALL) -m 0755 -d
+INSTALL_EXE    = $(INSTALL) -m 0755
+INSTALL_FILE   = $(INSTALL) -m 0644
+INSTALL_SECURE = $(INSTALL) -m 0600
+
 DESTDIR =
-ETC_DIRS = env.d
-LIB_DIRS = modprobe.d pam.d sysctl.d tmpfiles.d
-SHARE_DIRS = baselayout vim
 
-all: baselayout/shadow baselayout/gshadow
+all: lib/tmpfiles.d/baselayout-libdirs.conf share/baselayout/shadow share/baselayout/gshadow
+
+lib/tmpfiles.d/baselayout-libdirs.conf: Makefile
+	for d in $(LIBDIRS); do \
+		echo "L+	/$${d}	-	-	-	-	usr/$${d}"; \
+	done >$@
 
-baselayout/shadow: baselayout/passwd Makefile
+share/baselayout/shadow: share/baselayout/passwd Makefile
 	awk 'BEGIN {FS = ":"} { print $$1 ":*:15887:0:::::" }' <$< >$@
 
-baselayout/gshadow: baselayout/group Makefile
+share/baselayout/gshadow: share/baselayout/group Makefile
 	awk 'BEGIN {FS = ":"} { print $$1 ":*::" $$4 }' <$< >$@
 
 clean:
-	rm -f baselayout/shadow baselayout/gshadow
+	rm -f lib/tmpfiles.d/baselayout-libdirs.conf share/baselayout/shadow share/baselayout/gshadow
 
 install:
-	mkdir -m 0755 -p $(DESTDIR)/etc
-	cp -PR $(ETC_DIRS) $(DESTDIR)/etc
-	mkdir -m 0755 -p $(DESTDIR)/usr/lib
-	cp -PR $(LIB_DIRS) $(DESTDIR)/usr/lib
-	mkdir -m 0755 -p $(DESTDIR)/usr/share
-	cp -PR $(SHARE_DIRS) $(DESTDIR)/usr/share
+	if [[ -d bin ]]; then \
+		$(INSTALL_DIR) $(DESTDIR)/usr/bin; \
+		cp -pPR bin/* $(DESTDIR)/usr/bin/; \
+	fi
+	$(INSTALL_DIR) $(DESTDIR)/etc
+	cp -pPR etc/* $(DESTDIR)/etc/
+	$(INSTALL_DIR) $(DESTDIR)/usr/lib
+	cp -pPR lib/* $(DESTDIR)/usr/lib/
+	$(INSTALL_DIR) $(DESTDIR)/usr/share
+	cp -pPR share/* $(DESTDIR)/usr/share/
 	# no secrets in our shadow or sudoers but this is the proper way
 	chmod 0440 $(DESTDIR)/usr/share/baselayout/sudoers
 	chmod 0640 $(DESTDIR)/usr/share/baselayout/*shadow
-	# created by systemd's tmpfiles.d/tmp.conf but that is installed later
-	mkdir -m 1777 -p $(DESTDIR)/tmp $(DESTDIR)/var/tmp
 	# FHS compatibility symlinks stuff
 	ln -snf /var/tmp $(DESTDIR)/usr/tmp
 	# backwards compatibility for /etc/motd
 	ln -snf /run/flatcar/motd $(DESTDIR)/usr/share/baselayout/motd
 
-.PHONY: all clean install
+layout:
+	# bin directories
+	for d in bin local/bin; do \
+		$(INSTALL_DIR) $(DESTDIR)/usr/$${d} || exit 1; \
+		$(INSTALL_DIR) $(DESTDIR)/usr/lib/debug/usr/$${d} || exit 1; \
+	done
+	# lib directories
+	for d in $(LIBDIRS); do \
+		$(INSTALL_DIR) $(DESTDIR)/usr/$${d} $(DESTDIR)/usr/local/$${d} || exit 1; \
+		$(INSTALL_DIR) $(DESTDIR)/usr/lib/debug/usr/$${d} $(DESTDIR)/usr/lib/debug/usr/local/$${d} || exit 1; \
+	done
```

Comment:
```suggestion
	for d in $(foreach dir,bin $(LIBDIRS),$(dir) local/$(dir)); do \
		$(INSTALL_DIR) $(DESTDIR)/usr/$${d} || exit 1; \
		$(INSTALL_DIR) $(DESTDIR)/usr/lib/debug/usr/$${d} || exit 1; \
	done
```

Reply:
That's shorter indeed, but less readable. Also with `usr/local/sbin` *not* being a symlink to `bin`, but being a directory, this will need changes anyway.

Reply:
Somewhat used your suggestion but moved it to a separate make variable.